testgen/src/vote.rs

```diff
@@ -83,9 +89,13 @@ impl Generator<vote::Vote> for Vote {
         let signer = validator.get_private_key()?;
         let block_validator = validator.generate()?;
         let block_header = header.generate()?;
-        let block_id = block::Id {
-            hash: block_header.hash(),
-            part_set_header: PartSetHeader::new(1, block_header.hash()),
+        let block_id = if self.is_nil.is_some() {
+            None
+        } else {
+            Some(block::Id {
+                hash: block_header.hash(),
+                part_set_header: PartSetHeader::new(1, block_header.hash()),
+            })
```

Comment:
Do we want to generate `BlockIdFlagAbsent` votes as well?

Reply:
It does that automatically for validators who haven't voted ;) 
https://github.com/informalsystems/tendermint-rs/blob/3fb0cb9fcb0efae87dbcb813486992198e1aab8d/testgen/src/commit.rs#L146-L148